src/data.py

```diff
@@ -68,32 +67,66 @@ def dict(self) -> dict:
             for slotname in self.__slots__  # type: ignore
         }
 
-    def calculate_all(self, **kwargs):
-        """Calculate extra parameters and place them in `.extras` attribute and return self.extras."""
-        self.config = kwargs
+    def calculate_all(self,
+                      *,
+                      diffusing_element: str,
+                      known_structure: str | None = None,
+                      equilibration_steps: int = 1250,
+                      diffusion_dimensions: int = 3,
+                      z_ion: float = 1.0,
+                      n_parts: int = 10,
+                      dist_collective: float = 4.5):
+        """Calculate extra parameters and place them in `.extras` attribute and return self.extras.
 
-        equilibration_steps = kwargs.get('equilibration_steps')
-        diffusing_element = kwargs.get('diffusing_element')
+        Parameters
+        ----------
+        diffusing_element : str
+            Name of the diffusing element
+        structure : str | None
+            Path to cif file or name of known structure
+        equilibration_steps : int
+            Number of equilibration steps
+        diffusion_dimensions : int
+            Number of diffusion dimensions
+        z_ion : float
+            Ionic charge of the diffusing ion
+        n_parts : int
+            In how many parts to divide your simulation for statistics
+        dist_collective : float
+            Maximum distance for collective motions in Angstrom
+        """
+        extras = SimpleNamespace(
+            diffusing_element=diffusing_element,
+            known_structure=known_structure,
+            equilibration_steps=equilibration_steps,
+            diffusion_dimensions=diffusion_dimensions,
+            z_ion=z_ion,
+            n_parts=n_parts,
+            dist_collective=dist_collective,
+        )
+
+        self._add_shared_variables(extras)
+
+        extras.__dict__.update(Displacements.calculate_all(self,
+                                                           extras=extras))
+        extras.__dict__.update(Vibration.calculate_all(self, extras=extras))
+        extras.__dict__.update(Tracer.calculate_all(self, extras=extras))
```

Comment:
We could put everything in a dict here, and in the end create a namespace from it, instead of first creating the namespace, and then updating it as a dict

Reply:
Yeah, I considered that. I like the namespace because it's easier to type and we can change it for a (data)class at a later stage.